addon/-private/system/model/internal-model-class.js

```diff
@@ -0,0 +1,648 @@
+import Ember from 'ember';
+import {
+  relationshipFromMeta
+} from 'ember-data/-private/system/relationship-meta';
+import { assert, warn } from "ember-data/-private/debug";
+
+let COUNT_DATA = Object.create(null);
+
+function count(str) {
+  COUNT_DATA[str] = COUNT_DATA[str] ? COUNT_DATA[str] + 1 : 1;
+}
+
+window.getInternalModelClassCountData = function() {
```

Comment:
This should be removed. I've been using it for tracking when/where internally we use some of these hooks so that we can optimize them better and potentially deprecate a few of them.